v-next/hardhat/src/internal/builtin-plugins/network-manager/http-provider.ts

```diff
@@ -218,7 +225,8 @@ export class HttpProvider extends BaseProvider {
         throw new LimitExceededError(undefined, e);
       }
 
-      throw e;
+      // eslint-disable-next-line no-restricted-syntax -- allow throwing ProviderError
+      throw new UnknownError(e.message, e);
```

Comment:
This may report every HTTP-related error. We can start with reporting them, but we may need to filter most out later. LGTM though

Reply:
My thinking was to err on the side of reporting too much rather than too little.

Reply:
agreed